### PR TITLE
feat: EVM state verifier

### PR DESCRIPTION
**Pull request by hrls**

### CODE REVIEW

File: runtime/src/snapshot_utils.rs [hunk 1]
```diff
@@ -128,8 +128,8 @@ pub enum SnapshotError {
     #[error("accounts package send error")]
     AccountsPackageSendError(#[from] Box<AccountsPackageSendError>),
 
-    #[error("evm backup error: {0}")]
-    EvmBackupError(#[from] evm_state::storage::Error),
+    #[error("Wrong EVM state: {0}")]
+    EvmStateError(anyhow::Error),
```
Comment (vldm) [inline runtime/src/snapshot_utils.rs:132]: Why converting error from more concrete to abstract?

File: ci/docker-rust/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Note: when the rust version is changed also modify
 # ci/rust-version.sh to pick up the new image tag
-FROM rust:1.55.0
+FROM rust:1.56.1
```
Comment (vldm) [inline ci/docker-rust/Dockerfile]: Move version bump to separate pr

Comment (hrls) [reply ci/docker-rust/Dockerfile]: #247

File: runtime/src/snapshot_utils.rs (hunk 1 above)
Comment (hrls) [reply runtime/src/snapshot_utils.rs:132]: Specific error type in this variant should be union of `evm_state::storage::Error` and return type of func calls from `evm_state::storage::walker::*`. The last one is `anyhow::Error`. So we need introduce additional error type for walker and sum it with storage error. Just for print it on stderr. There is no recover anyway.

**Comment by hrls** [comment]
Based on #245

**Review by vldm — APPROVED** [review]

**Review by vldm — APPROVED** [review]
